examples/GLUE/src/train.py

```diff
@@ -0,0 +1,368 @@
+        return {
+            "pearson": pearson_corr,
+            "spearmanr": spearman_corr,
+            "corr": (pearson_corr + spearman_corr) / 2,
+        }
+        
+    def on_end_val_epoch(
+        self, global_step: int, *val_step_collated_outputs, key="default"
+    ):
+        """
+        callback after validation loop ends
+        """
+        loss, predicted, labels = val_step_collated_outputs
+        # print(labels, predicted)
+        loss = loss.mean().item()
+        correlations = self.pearson_and_spearman(labels, predicted)
+        #https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.spearmanr.html
+        global_stats.update(f"{self.glue_task}/val_{key}/loss", loss)
+        global_stats.update_multi(f"{self.glue_task}/val_{key}", correlations)
+    
+class MNLIClassifier(SentencePairClassifier):
+    def __init__(self, *args, max_batches_per_epoch = 500 , **kwargs):
+        super().__init__(*args,**kwargs)
+        self.max_batches_per_epoch = max_batches_per_epoch
+        self.dl = None
+
+    
+    def get_val_dataloaders(self, sampler: torch.utils.data.Sampler, batch_size: int):
+        return {key:torch.utils.data.DataLoader(
+                ds,
+                sampler = sampler(ds),
+                batch_size=batch_size,
+                collate_fn=self.collate_fun,
+            ) for key,ds in self.data_interface.get_val_dataset().items()}
+
+    def on_end_val_epoch(
+        self, global_step: int, *val_step_collated_outputs, key="default"
+    ):
+        """
+        callback after validation loop ends
+        """
+        loss, predicted, labels = val_step_collated_outputs
+        # print(labels, predicted)
+        loss = loss.mean().item()
+        acc = accuracy_score(labels, predicted)
+        global_stats.update(f"{self.glue_task}/val_{key}/loss", loss)
+        global_stats.update(f"{self.glue_task}/val_{key}/acc", acc)
+        # global_stats.log_model(global_step, self, force = True, grad_scale=1)
+
+def recipe_factory(glue_task, *args, **kwargs):
+    factory = {
+        'cola':SentenceClassifier,
+        'qqp':SentencePairClassifier,
+        'rte':SentencePairClassifier,
+        'qnli':SentencePairClassifier,
+        'mnli':MNLIClassifier,
+        'sst2':SentenceClassifier,
+        'stsb':STSBPairClassifier,
+        'wnli':SentencePairClassifier,
+        'mrpc':SentencePairClassifier,
+    }
+    return factory[glue_task](*args, **kwargs)
+    
+from data import data_factory
+
+def run_glue_finetune(config):
+
+    glue_task = config['glue_task']
+    print(glue_task)
+    data = data_factory(glue_task)
+    data.setup_datasets(glue_task)
+    print(data.get_train_dataset()[:5])
+
+    recipe = recipe_factory(glue_task, data_interface = data, **config['mi'])
+
+    # Trianing code
+
+    trainer_backend = SingleProcessAmp() #DDPTrainerBackend(SingleProcessAmp())
+    distributed_training_args= DistributedTrainingArguments(**config["dist"])
```

Comment:
we have trainer backend factory in trainer that will get called if you set trainer.backend in config. no need to create a dist and distributed parameter group to be handled seperately in scenario.

Reply:
https://github.com/microsoft/PyMarlin/blob/43dc8389d1e1a122c301aa1d905fed470ff6b931/pymarlin/core/trainer.py#L233